lib/Service/CustomerService.php

```diff
@@ -0,0 +1,267 @@
+<?php
+
+namespace Stripe\Service;
+
+class CustomerService extends AbstractService
+{
+    const BALANCE_TRANSACTIONS = "balance_transactions";
+    const SOURCES = "sources";
+    const TAX_IDS = "tax_ids";
+
+    public function basePath()
+    {
+        return "/v1/customers";
+    }
+
+    /**
+     * List all customers.
+     *
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\Collection
+     */
+    public function all($params = [], $opts = [])
+    {
+        return $this->allObjects($params, $opts);
+    }
+
+    /**
+     * List customer balance transactions.
+     *
+     * @param string $customerId
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\Collection
+     */
+    public function allBalanceTransactions($customerId, $params = [], $opts = [])
+    {
+        return $this->allNestedObjects(self::BALANCE_TRANSACTIONS, $customerId, $params, $opts);
+    }
+
+    /**
+     * List all sources.
+     *
+     * @param string $customerId
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\Collection
+     */
+    public function allSources($customerId, $params = [], $opts = [])
+    {
+        return $this->allNestedObjects(self::SOURCES, $customerId, $params, $opts);
+    }
+
+    /**
+     * List all tax IDs.
+     *
+     * @param string $customerId
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\Collection
+     */
+    public function allTaxIds($customerId, $params = [], $opts = [])
+    {
+        return $this->allNestedObjects(self::TAX_IDS, $customerId, $params, $opts);
+    }
+
+    /**
+     * Create a customer.
+     *
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\Customer
+     */
+    public function create($params = [], $opts = [])
+    {
+        return $this->createObject($params, $opts);
+    }
+
+    /**
+     * Create a balance transaction.
+     *
+     * @param string $customerId
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\CustomerBalanceTransaction
+     */
+    public function createBalanceTransaction($customerId, $params = [], $opts = [])
+    {
+        return $this->createNestedObject(self::BALANCE_TRANSACTIONS, $customerId, $params, $opts);
+    }
+
+    /**
+     * Create a source.
+     *
+     * @param string $customerId
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\Account|\Stripe\AlipayAccount|\Stripe\BankAccount|\Stripe\BitcoinReceiver|\Stripe\Card|\Stripe\Source
+     */
+    public function createSource($customerId, $params = [], $opts = [])
+    {
+        return $this->createNestedObject(self::SOURCES, $customerId, $params, $opts);
+    }
+
+    /**
+     * Create a tax ID.
+     *
+     * @param string $customerId
+     * @param array $params
+     * @param array $opts
+     * @return \Stripe\TaxId
```

Comment:
We likely need to talk more about whether to prefix by the parent resource name or not, so TaxId versus CustomerTaxId. Unrelated to your PR, just something to think about

Reply:
Tax IDs have an `object` key with value `tax_id`, so I think `TaxId` is the appropriate name even if they are always nested under customers.